app/controllers/appointments_controller.rb

```diff
@@ -18,14 +20,35 @@ def create
     created_appointment
   end
 
+  def destroy
+    if @appointment.present?
+      @appointment.destroy
+      flash[:success] = t "appointment_canceled"
+      redirect_to request.referer || root_url
+    else
+      flash[:danger] = t "not_success"
+      redirect_to root_url
+    end
+  end
+
+  def update
+    if @appointment.update appointment_params
+      flash[:success] = t "appointment_accepted"
+    else
+      flash[:danger] = t "appointment_not_accepted"
+    end
+    redirect_to appointments_path
+  end
+
   private
 
   def appointment_params
-    params.require(:appointment).permit Appointment::APPOINTMENT_PARAMS
+    ap = params.require(:appointment).permit Appointment::APPOINTMENT_PARAMS
+    ap[:status] = params[:appointment][:status].to_i
+    return ap
```

Comment:
(kill)

Reply:
là sao anh? . Em ban đầu truyền luôn cái status vào cái tham số thì khi em dùng value bên cái button accept . nếu em để value: 1 thì nó sẽ báo lỗi bảo '1' is not a valid status. Nên em phải làm vậy để nó nhận 1 ạ